packages/ra-ui-materialui/src/list/filter/FilterListItem.tsx

```diff
@@ -144,27 +145,30 @@ const useStyles = makeStyles(theme => ({
  */
 const FilterListItem: FC<{ label: string; value: any }> = props => {
     const { label, value } = props;
-    const { filterValues, setFilters } = useListFilterContext();
+    const {
+        filterValues,
+        setFilters,
+        displayedFilters,
+        hideFilter,
+    } = useListContext();
     const translate = useTranslate();
     const classes = useStyles(props);
 
-    const isSelected = isEqual(value, filterValues);
+    const isSelected =
+        isEmpty(value) || isEmpty(filterValues)
+            ? false
+            : isMatch(filterValues, value);
 
     const addFilter = () => {
-        setFilters({ ...filterValues, ...value }, null, false);
+        setFilters(merge({}, filterValues, value), null, false);
```

Comment:
Problem : we should update URL properly with the nested fields of value and without overriding nodes of other filters contained in filterValues.
Correction : using lodash deep merge instead of spread operator.